docs/problem_format/cpp_psetting_templates.md

```diff
@@ -0,0 +1,56 @@
+# C++ Problem Setting Templates - `cpp_psetting_templates`
+
+There are three C++ input-handling templates provided for aiding problem setters. They are as follows:
+
+- [Validator Template](https://github.com/DMOJ/docs/blob/master/sample_files/problem_setting/validator.cpp)
+- [Identical Checker/Interactor Template](https://github.com/DMOJ/docs/blob/master/sample_files/problem_setting/identical_checker_interactor.cpp)
+- [Standard Checker/Interactor Template](https://github.com/DMOJ/docs/blob/master/sample_files/problem_setting/standard_checker_interactor.cpp)
+
+## Validator
+
+This is a template for validating the input data of problems. It aims to be simple and of course, correct. It contains 6 functions. The first three are whitespace functions:
+
+- `void readSpace()` expects a space at the current position in the input, and aborts the program if there is not a space.
+- `void readNewLine()` expects a newline at the current position in the input.
+- `void readEOF()` expects the input file to end immediately at the current position.
+
+The remaining three are for actual content:
+
+- `std::string readToken(char min_char = 0, char max_char = 127)` returns the next token in the input stream. A token is defined as a whitespace-separated string. If the next character in the input is a whitespace character, this method aborts the program. The optional arguments `min_char` and `max_char` can be used to enforce a range on the characters in the token. For instance, `readToken('a', 'z')` reads a lowercase string of english letters.
```

Comment:
I think we should also have a `readLine` method, in case a particular line of input is very complex and the user wants to validate that themselves but otherwise your template

Reply:
Sure - how should this interact with the standard checker? We can also choose to not offer this method under the standard checker.

Reply:
I think it should be provided with the standard checker, but with a warning that it might not do what is obvious (in case weird parsing is needed).

Reply:
I think `readLine` is strictly incorrect under standard checker, always, and so I won't offer it under the standard checker.